subprojects/docs/src/docs/userguide/java_plugin.adoc

```diff
@@ -41,6 +41,71 @@ include::javaProjectTestLayout.adoc[]
+
+`(read-only) File testResultsDir`::
+The directory to generate test result .xml files into. Default value: `__buildDir__/__testResultsDirName__`
+
+`String testReportDirName`::
+The name of the directory to generate the test report into, relative to the reports directory. Default value: `tests`
+
+`(read-only) File testReportDir`::
+The directory to generate the test report into. Default value: `__reportsDir__/testReportDirName`
+
+`String libsDirName`::
+The name of the directory to generate libraries into, relative to the build directory. Default value: `libs`
+
+`(read-only) File libsDir`::
+The directory to generate libraries into. Default value: `__buildDir__/__libsDirName__`
+
+`String distsDirName`::
+The name of the directory to generate distributions into, relative to the build directory. Default value: `distributions`
+
+`(read-only) File distsDir`::
+The directory to generate distributions into. Default value: `__buildDir__/__distsDirName__`
+
+`String docsDirName`::
+The name of the directory to generate documentation into, relative to the build directory. Default value: `docs`
+
+`(read-only) File docsDir`::
+The directory to generate documentation into. Default value: `__buildDir__/__docsDirName__`
+
+`String dependencyCacheDirName`::
+The name of the directory to use to cache source dependency information, relative to the build directory. Default value: `dependency-cache`
+
+=== Other convention properties
+
+`(read-only) link:{javadocPath}/org/gradle/api/tasks/SourceSetContainer.html[SourceSetContainer] sourceSets`::
+Contains the project's source sets. Default value: Not null link:{javadocPath}/org/gradle/api/tasks/SourceSetContainer.html[SourceSetContainer]
+
+`link:{javadocPath}/org/gradle/api/JavaVersion.html[JavaVersion] sourceCompatibility`::
+Java version compatibility to use when compiling Java source. Default value: version of the current JVM in use link:{javadocPath}/org/gradle/api/JavaVersion.html[JavaVersion]. Can also set using a String or a Number, e.g. `'1.5'` or `1.5`.
+
+`link:{javadocPath}/org/gradle/api/JavaVersion.html[JavaVersion] targetCompatibility`::
+Java version to generate classes for. Default value: `__sourceCompatibility__`. Can also set using a String or Number, e.g. `'1.5'` or `1.5`.
+
+`String archivesBaseName`::
+The basename to use for archives, such as JAR or ZIP files. Default value: `__projectName__`
+
+`link:{javadocPath}/org/gradle/api/java/archives/Manifest.html[Manifest] manifest`::
+The manifest to include in all JAR files. Default value: an empty manifest.
+
+These properties are provided by convention objects of type link:{javadocPath}/org/gradle/api/plugins/JavaPluginConvention.html[JavaPluginConvention], and link:{javadocPath}/org/gradle/api/plugins/BasePluginConvention.html[BasePluginConvention].
+
 [[sec:changing_java_project_layout]]
 === Changing the project layout
```

Comment:
This section now looks out of place. I think that's partly because it's explanatory text that should probably be in the main part of the user manual rather than in a reference chapter. I would consider putting this into the _Building Java & JVM projects_ chapter, with links back to the convention properties and source set properties that allow you to fine tune the project layout. What do you think?

Reply:
The above section on the default project layout could then link to the content in _Building Java & JVM projects_, similarly to how the _Defining new source sets_ section does it below.